providers/cnr/CNRProvider.go

```diff
@@ -0,0 +1,81 @@
+// Package CNR implements a registrar that uses the CNR api to set name servers. It will self register it's providers when imported.
+package cnr
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/StackExchange/dnscontrol/v4/providers"
+	cnrcl "github.com/centralnicgroup-opensource/rtldev-middleware-go-sdk/v5/apiclient"
+)
+
+// GoReleaser: version
+var (
+	version = "dev"
+)
+
+// CNRClient describes a connection to the CNR API.
+type CNRClient struct {
+	APILogin    string
+	APIPassword string
+	APIEntity   string
+	client      *cnrcl.APIClient
+}
+
+var features = providers.DocumentationNotes{
+	// The default for unlisted capabilities is 'Cannot'.
+	// See providers/capabilities.go for the entire list of capabilities.
+	providers.CanGetZones:            providers.Unimplemented(),
+	providers.CanConcur:              providers.Cannot(),
```

Comment:
Suggestion: Try setting CanConcur to providers.Can().  I don't see any potential concurrency issues.  You aren't storing any state in the CNRClient that changes after initialization.

To test this, enable the race condition detector and use a dnsconfig.js with at least 3 domains (more is better)

$ cd ~/git/dnscontrol
$ go install -race
$ cd (wherever your dnsconfig is)
$ dnscontrol preview

If you don't get any errors, CanConcur() set to Can() is recommended.

Reply:
No errors, worked like a charm. Thanks!